src/ensembl/src/shared/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+
+import defaultStyles from './Checkbox.scss';
+
+export enum CheckboxStatus {
+  CHECKED = 'checked',
+  DISABLED = 'disabled',
+  UNCHECKED = 'unchecked'
+}
+
+type Props = {
+  onClick?: () => void;
+  status: CheckboxStatus;
+  classNames?: { [key in CheckboxStatus]: string };
+  label?: string;
+  labelClassName?: string;
```

Comment:
labelClassName is useless without the label. If label is optional, then perhaps label and labelClassName should be combined in a single type (so that if there is no label, then the component should not accept the  labelClassName prop)?

```
type WithoutLabelProps = {
  status: ...,
  onChange: ...,
}

type WithLabelProps = WithoutLabelProps & {
  label: ...
  labelClassName?: ...
}

type Props = WithoutLabelProps | WithLabelProps;
```

Reply:
That's done. 
TypeScript is not happy for some reason as it says:
```
Property 'label' does not exist on type 'Props'.
  Property 'label' does not exist on type 'WithoutLabelProps'.
```

Reply:
@msmanoj That is because `WithLabelProps` may not exist. You will need add some checks before using `label` and `labelClassName`.

Reply:
This should be resolved by 48819b6